### ISSUE TITLE
Little changes in SecHub report summary

### ISSUE THREAD

**Issue by de-jcup** [mercedes-benz/sechub#2679]
## Situation

We have a very good contribution in PR #2156 to improve SecHub's reporting options.

In terms of content, everything desired was implemented and the review PR was delayed for a long time - thanks again to @zigfridus for his patience.

However, there were still a few little things that didn't fit our usual code architecturally, or where e.g. class names are not immediately clear.

In order not to hold up the PR unnecessarily - the content was fine - this issue was created to collect these little things and deal with them afterwards.

<sup>PS: We should perhaps generally choose this approach for longer PRs in order to avoid unnecessary frustration among contributors.<sup>

## Solution
- Collect the necessary changes inside this issue
- Implement the changes after PR #2156 has been merged

**Comment by de-jcup** [comment]
## Wanted changes
### `SecHubReportSummary`  

  was new introduced, but is inside `SecHubReportMetaData`
  There exists a class `SecHubReportMetaDataSummary` but it seems it is one level lower

- [ ] rename the class, if necessary rename the subclasses (but do not change the model)

### SecHubReportMetaDataSummary
- [ ] contains long total = 0 ; -> unnecessary in java, literal can be removed because 0 is default
- [ ] Move logic to other location  
      (the json data classes shall contain no logic when possible)
         - reportScanHelper  
           (remark a method name should also describe what it does - e.g. "fillSeverityDetails")
         - incrementColors and increment methods 
         
### SecHubReportMetaDataSummaryDetails
- [ ] Move logic to other location (the json data classes shall contain no logic when possible)
         - detailsHelper   (remark a method name should also describe what it does - e.g. "fillDetails")
         - detailsFiller and increment methods
- [ ] Improve log handling
  LOG.debug for exceptions -> an error should not be treated this way. is the model still valid? Logic must be checked here as well.
  LOG statements with parameters check
- [ ] TreeMapDeserializer
  Hmm. sounded first like it could be its own class and very generic.
  But seems to be very specialized.
  Either Move in commons and use generics or rename it and keep it private.

**Comment by zigfridus** [comment]
@de-jcup 
Thank you for reviewing my code in #2156. I will correct the flaws you found after I finish my work on the issue I am currently working on.

**Comment by de-jcup** [comment]
@zigfridus: I would do it the following way: Currently I am still reviewing #2156 - Unfortunately I found some bugs. In the PR #2156 only the bugfix is necessary. Means report is working and has no errors - changes can be done later.

When merge is done, I would start implementing this issue (#2679) by myself. When I have finished and it is reviewed by other core members, I would inform you. By doing this, you would be able to look at my changes inside my PR. Having those changes and this issue you will see why and what I did change. I think this is easier, gives you hints/ideas and is less frustrating for yourself.

If you would like we could also make an virtual appointment where I could explain the changes etc.
